packages/qwik/src/core/v2/client/vnode.ts

```diff
@@ -1004,11 +1072,180 @@ export const vnode_insertBefore = (
   newChild[VNodeProps.parent] = parent;
 };
 
+function vnode_getDomChildrenToInsert(
+  journal: VNodeJournal,
+  document: Document,
+  domParentVNode: ElementVNode,
+  newChild: VNode
+) {
+  const parentNodeIsDefaultNamespace = vnode_isDefaultNamespace(domParentVNode);
+  const parentIsForeignObject = !parentNodeIsDefaultNamespace
+    ? isForeignObjectElement(vnode_getElementName(domParentVNode))
```

Comment:
here is the same problem with element name

Reply:
This seems like a lot of code. Why is all of this needed? Also, can we move all of the code that morphs from one namespace into a different namespace into a separate file?